api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -961,54 +961,40 @@ public DbUser getByUsernameOrThrow(String username) {
 
   @Override
   public void updateBypassTime(long userDatabaseId, AccessBypassRequest accessBypassRequest) {
-    final DbUser user =
-        getByDatabaseId(userDatabaseId)
-            .orElseThrow(
-                () ->
-                    new NotFoundException(
-                        String.format("User with database ID %d not found", userDatabaseId)));
+    getByDatabaseId(userDatabaseId)
+        .orElseThrow(
+            () ->
+                new NotFoundException(
+                    String.format("User with database ID %d not found", userDatabaseId)));
 
     final Timestamp newBypassTime;
-    final Timestamp previousBypassTime;
 
     final Boolean isBypassed = accessBypassRequest.getIsBypassed();
     if (isBypassed) {
       newBypassTime = new Timestamp(clock.instant().toEpochMilli());
     } else {
       newBypassTime = null;
     }
+
     switch (accessBypassRequest.getModuleName()) {
       case DATA_USE_AGREEMENT:
```

Comment:
I missed the previous value in my original implementation. Let's keep these and one of us can pass the previous time as an additional Optional<Long> arg to `fireAdministrativeBypassTime()`. You can just do a TODO or go ahead and pipe that in if you don't mind.

Reply:
OK, added it